getdata_HN.py

```diff
@@ -8,56 +8,54 @@
 URL = 'https://hacker-news.firebaseio.com/v0/'
 KEYWORDS = ['python', 'django']
 DATAFILE = 'data.json'
-stories = {}
 
 
 def _get_json(param):
     url = urljoin(URL, param)
-    res = requests.get(url)
+    try:
+        res = requests.get(url)
+    except requests.exceptions.ConnectionError as err:
+        print('Err {0}'.format(err))
     return res.json()
 
 
-def _get_stories():
-    parameter = 'topstories.json?print=pretty'
-    return _get_json(parameter)
-
-
-def get_stories_of_topics():
-    for story in _get_stories():
-        parameter = 'item/{}.json?print=pretty'.format(story)
-        story = _get_json(parameter)
+def _get_stories_of_topics(topics=None):
+    stories = {}
+    param_ids = 'topstories.json?print=pretty'
+    stories_id = _get_json(param_ids)
+    for story_id in stories_id:
+        param_id = 'item/{}.json?print=pretty'.format(story_id)
+        story = _get_json(param_id)
+        story_title = story.get('title', 'None')
         for kwd in KEYWORDS:
-            if kwd in story['title'].lower():
-                if 'url' in story:
-                    url_story = story['url']
-                else:
-                    url_story = '#'
-                stories[story['id']] = {
-                    'title': story['title'],
-                    'url': url_story
+            if kwd in story_title.lower():
+                story_url = story.get('url', '#')
+                stories[str(story_id)] = {
+                    'title': story_title,
+                    'url': story_url
                 }
     return stories
 
 
-def _save_stories(all_stories):
+def _save_stories(stories):
     with open(DATAFILE, 'w') as outfile:
-        json.dump(all_stories, outfile, indent=4)
+        json.dump(stories, outfile, indent=4)
     return
 
 
 def main():
     if os.path.isfile(DATAFILE):
```

Comment:
this condition is "too broad".

You can just try load data (try/except), if it failed, set old_stories = {}

because logic for both case are the same